src/components/modal-flow/ModalFlow.js

```diff
@@ -0,0 +1,373 @@
+            styles.logoWrapper,
+            !this.isAnimatingLayout && styles.visible
+        );
+
+        return (
+            <div className={ logoWrapperClasses }>
+                { this.state.layout !== null && <Logo className={ styles.logo } /> }
+            </div>
+        );
+    };
+
+    inferLayoutTransition = () => {
+        const { layout, requestNextLayout } = this.state;
+
+        switch (layout) {
+        case null:
+            if (requestNextLayout === LAYOUT.HALF || LAYOUT.HALF_BORDERED) {
+                this.isAnimatingLayout = true;
+
+                return LAYOUT_TRANSITION.EMPTY_TO_HALF;
+            }
+            break;
+        case LAYOUT.HALF:
+            if (this.prevLayout === LAYOUT.FULL && this.isAnimatingLayout) {
+                this.stepsPlacement = 'right';
+
+                return LAYOUT_TRANSITION.FULL_TO_HALF_ENTERING;
+            }
+            if (requestNextLayout === LAYOUT.FULL) {
+                this.isAnimatingLayout = true;
+
+                return LAYOUT_TRANSITION.HALF_TO_FULL_EXITING;
+            }
+            break;
+        case LAYOUT.FULL:
+            if (requestNextLayout === LAYOUT.HALF) {
+                this.isAnimatingLayout = true;
+
+                return LAYOUT_TRANSITION.FULL_TO_HALF_EXITING;
+            }
+            if (this.prevLayout === LAYOUT.HALF && this.isAnimatingLayout) {
+                this.stepsPlacement = 'left';
+
+                return LAYOUT_TRANSITION.HALF_TO_FULL_ENTERING;
+            }
+            if (this.prevLayout === LAYOUT.WIDE && this.isAnimatingLayout) {
+                return LAYOUT_TRANSITION.WIDE_TO_FULL_ENTERING;
+            }
+            if (requestNextLayout === LAYOUT.WIDE) {
+                this.isAnimatingLayout = true;
+
+                return LAYOUT_TRANSITION.FULL_TO_WIDE_EXITING;
+            }
+            break;
+        case LAYOUT.HALF_BORDERED:
+            if (requestNextLayout === LAYOUT.WIDE) {
+                this.isAnimatingLayout = true;
+
+                return LAYOUT_TRANSITION.HALF_BORDERED_TO_WIDE_EXITING;
+            }
+            break;
+        case LAYOUT.WIDE:
+            if (this.prevLayout === LAYOUT.HALF_BORDERED && this.isAnimatingLayout) {
+                return LAYOUT_TRANSITION.HALF_BORDERED_TO_WIDE_ENTERING;
+            }
+            if (layout === LAYOUT.WIDE && requestNextLayout === LAYOUT.FULL) {
+                this.isAnimatingLayout = true;
+
+                return LAYOUT_TRANSITION.WIDE_TO_FULL_EXITING;
+            }
+            if (this.prevLayout === LAYOUT.FULL && this.isAnimatingLayout) {
+                return LAYOUT_TRANSITION.FULL_TO_WIDE_ENTERING;
+            }
+            break;
+        default: return null;
+        }
+    };
+
+    handleCurrentStepTransitionEnd = (event) => {
+        const { requestNextStepIndex } = this.state;
+
+        // Ignore `transition end` event for width property
+        if (event.propertyName === 'width') {
```

Comment:
We should make sure that this event is coming from the correct `target` because this event bubbles and might be coming from a deep child. Perhaps, by doing that, we won't need to compare the `propertyName`.

Reply:
There is a specific case where I need to be aware of a deeper child event. The scenario is the following:
When `wide` layout is applied, steps are inside a block and a `transform` property is applied on a specific step `div`. Also, a width transition is also applied but I don't need to be aware of it. That is the reason why I'm doing this.

Is there any better way to do this?